File: php/features.php
<?php
declare(strict_types=1);

// RemoteJobsApiRssFeed SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RemoteJobsApiRssFeedFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RemoteJobsApiRssFeedBaseFeature();
            case "test":
                return new RemoteJobsApiRssFeedTestFeature();
            default:
                return new RemoteJobsApiRssFeedBaseFeature();
        }
    }
}

File: py/remotejobsapirssfeed_sdk/features.py
# RemoteJobsApiRssFeed SDK feature factory

from remotejobsapirssfeed_sdk.feature.base_feature import RemoteJobsApiRssFeedBaseFeature
from remotejobsapirssfeed_sdk.feature.test_feature import RemoteJobsApiRssFeedTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RemoteJobsApiRssFeedBaseFeature(),
        "test": lambda: RemoteJobsApiRssFeedTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
